File: CS/DXApplication1/Main.cs
using DevExpress.XtraBars.Ribbon;
using DXApplication.Data;

namespace DXApplication {
    public partial class Main : RibbonForm {
        public Main() {
            InitializeComponent();

            gridView.OptionsView.RowAutoHeight = true;
        }
        async void BindToDataTable(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            gridView.LoadingPanelVisible = true;
            gridControl.DataSource = await DataTableProvider.Instance.GetDataAsync();
            gridView.BestFitColumns();
            gridView.LoadingPanelVisible = false;
        }
        async void BindToSqlDataSource(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            gridView.LoadingPanelVisible = true;
            gridControl.DataSource = await SqlDataSourceProvider.Instance.GetDataAsync();
            gridControl.DataMember = "AzureQuery";
            gridView.BestFitColumns();
            gridView.LoadingPanelVisible = false;
        }
        async void BindToEF(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            gridView.LoadingPanelVisible = true;
            gridControl.DataSource = await EFDataProvider.Instance.GetDataAsync();
            gridView.BestFitColumns();
            gridView.LoadingPanelVisible = false;
        }
    }
}

File: VB/DXApplication1/Main.vb
Imports DevExpress.XtraBars.Ribbon
Imports DXApplication.Data

Namespace DXApplication
	Partial Public Class Main
		Inherits RibbonForm

		Public Sub New()
			InitializeComponent()

			gridView.OptionsView.RowAutoHeight = True
		End Sub
		Private Async Sub BindToDataTable(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles useDataTabletem.ItemClick
			gridView.LoadingPanelVisible = True
			gridControl.DataSource = Await DataTableProvider.Instance.GetDataAsync()
			gridView.BestFitColumns()
			gridView.LoadingPanelVisible = False
		End Sub
		Private Async Sub BindToSqlDataSource(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles useSqlDataSourceItem.ItemClick
			gridView.LoadingPanelVisible = True
			gridControl.DataSource = Await SqlDataSourceProvider.Instance.GetDataAsync()
			gridControl.DataMember = "AzureQuery"
			gridView.BestFitColumns()
			gridView.LoadingPanelVisible = False
		End Sub
		Private Async Sub BindToEF(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles useEFItem.ItemClick
			gridView.LoadingPanelVisible = True
			gridControl.DataSource = Await EFDataProvider.Instance.GetDataAsync()
			gridView.BestFitColumns()
			gridView.LoadingPanelVisible = False
		End Sub
	End Class
End Namespace
